client/state/hosting/reducer.js

```diff
@@ -0,0 +1,48 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import { combineReducers } from 'state/utils';
+import { HOSTING_SFTP_USER_UPDATE } from 'state/action-types';
+
+/**
+ * Responsible for the stfp user details for Atomic sites. Currently this handles
+ * the current admin user only, but is structured to be able to handle all sftp
+ * user details for a given site in the future releases of this feature.
+ * atomicHosting: {
+ *      [ siteId ]: {
+ *          [ sftpUsers ]: {
+ *              [ userId ]: {
+ *                  userName: 'smithers-jones',
+ *                  password: 'super-secret'
+ *              }
+ *          }
+ *      }
+ * }
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+const sites = ( state = {}, action ) => {
+	const { siteId, userId } = action;
+	const site = state[ siteId ] ? state[ siteId ] : {};
```

Comment:
Could we use `keyedReducer` here?

```js
import { HOSTING_SFTP_USER_UPDATE } from 'state/action-types';
import { keyedReducer } from 'state/utils';

export const atomicHosting = ( state, { type, { userId, sftpUser } } ) =>
	type === HOSTING_SFTP_USER_UPDATE 
		? { sftpUsers: ...state, [ userId ]: sftpUser }
		: state;

export default keyedReducer( 'siteId', atomicHosting );
```

Reply:
Another handy tip - I had to use a slightly different setup to get it working, but still much simpler than my original.